### PR TITLE
Introduce anthropic plugin

### PR DESCRIPTION
**Pull request by keepingitneil**

### CODE REVIEW

**Review by nbsp — APPROVED** [review]
weak LGTM, probably wait for another review but nothing jumps out

File: .github/workflows/tests.yml
Comment (theomonnom) [inline .github/workflows/tests.yml:79]: let's also add the plugin to the type-checks CI

File: tests/test_llm.py
Comment (theomonnom) [inline tests/test_llm.py:221]: do we need the job_process fixture for the LLMs?

File: livekit-plugins/livekit-plugins-anthropic/livekit/plugins/anthropic/llm.py
Comment (theomonnom) [inline livekit-plugins/livekit-plugins-anthropic/livekit/plugins/anthropic/llm.py:66]: This seems unused

File: livekit-plugins/livekit-plugins-anthropic/livekit/plugins/anthropic/llm.py
Comment (theomonnom) [inline livekit-plugins/livekit-plugins-anthropic/livekit/plugins/anthropic/llm.py:446]: Seems like there is no change compared to https://github.com/livekit/agents/blob/80dac5600bcc1c0863b98982a626e201f1be97a4/livekit-agents/livekit/agents/llm/_oai_api.py#L143

**Review by theomonnom — APPROVED** [review]
lgtm

File: livekit-plugins/livekit-plugins-anthropic/livekit/plugins/anthropic/llm.py
Comment (theomonnom) [inline livekit-plugins/livekit-plugins-anthropic/livekit/plugins/anthropic/llm.py:187]: I think this isn't needed

File: livekit-plugins/livekit-plugins-anthropic/livekit/plugins/anthropic/llm.py
Comment (theomonnom) [inline livekit-plugins/livekit-plugins-anthropic/livekit/plugins/anthropic/llm.py:88]: nit: I would call that _merge_messages but no strong preference

File: livekit-plugins/livekit-plugins-anthropic/livekit/plugins/anthropic/llm.py
Comment (keepingitneil) [reply livekit-plugins/livekit-plugins-anthropic/livekit/plugins/anthropic/llm.py:187]: I think it is no? Because it is in `__anext__`

File: livekit-plugins/livekit-plugins-anthropic/livekit/plugins/anthropic/llm.py
Comment (keepingitneil) [reply livekit-plugins/livekit-plugins-anthropic/livekit/plugins/anthropic/llm.py:446]: yeah I can put this in a common utils in another PR. I'll hold off for now because tests are flaky (due to 429s)

**Pull request merged**
